packages/lambdas/src/shared/bundle-to-html.ts

```diff
@@ -0,0 +1,1181 @@
+          ${createObservationLaboratorySection(observationLaboratory)}
+          ${createEncountersSection(encounters)}
+          ${createImmunizationSection(immunizations)}
+          ${createFamilyHistorySection(familyMemberHistories)}
+          ${createRelatedPersonSection(relatedPersons)}
+          ${createTaskSection(tasks)}
+          ${createCoverageSection(coverages)}
+        </div>
+        <div id="mr-footer"></div>
+      </body>
+    </html>
+  `;
+
+  return htmlPage;
+};
+
+function extractFhirTypesFromBundle(bundle: Bundle): {
+  patient?: Patient;
+  medications: MedicationRequest[];
+  conditions: Condition[];
+  allergies: AllergyIntolerance[];
+  procedures: Procedure[];
+  observationSocialHistory: Observation[];
+  observationVitals: Observation[];
+  observationLaboratory: Observation[];
+  encounters: Encounter[];
+  immunizations: Immunization[];
+  familyMemberHistories: FamilyMemberHistory[];
+  relatedPersons: RelatedPerson[];
+  tasks: Task[];
+  coverages: Coverage[];
+} {
+  let patient: Patient | undefined;
+  const medications: MedicationRequest[] = [];
+  const conditions: Condition[] = [];
+  const allergies: AllergyIntolerance[] = [];
+  const procedures: Procedure[] = [];
+  const observationSocialHistory: Observation[] = [];
+  const observationVitals: Observation[] = [];
+  const observationLaboratory: Observation[] = [];
+  const encounters: Encounter[] = [];
+  const immunizations: Immunization[] = [];
+  const familyMemberHistories: FamilyMemberHistory[] = [];
+  const relatedPersons: RelatedPerson[] = [];
+  const tasks: Task[] = [];
+  const coverages: Coverage[] = [];
+
+  if (bundle.entry) {
+    for (const entry of bundle.entry) {
+      const resource = entry.resource;
+      if (resource?.resourceType === "Patient") {
+        patient = resource as Patient;
+      } else if (resource?.resourceType === "MedicationRequest") {
+        medications.push(resource as MedicationRequest);
+      } else if (resource?.resourceType === "Condition") {
+        conditions.push(resource as Condition);
+      } else if (resource?.resourceType === "AllergyIntolerance") {
+        allergies.push(resource as AllergyIntolerance);
+      } else if (resource?.resourceType === "Procedure") {
+        procedures.push(resource as Procedure);
+      } else if (resource?.resourceType === "Observation") {
+        const observation = resource as Observation;
+        const stringifyResource = JSON.stringify(resource);
+
+        if (stringifyResource && stringifyResource.toLowerCase().includes("vital signs")) {
+          observationVitals.push(observation);
+        } else if (stringifyResource && stringifyResource.toLowerCase().includes("laboratory")) {
+          observationLaboratory.push(observation);
+        } else if (
+          stringifyResource &&
+          stringifyResource.toLowerCase().includes("social history")
+        ) {
+          observationSocialHistory.push(observation);
+        }
```

Comment:
1. we may want to consider having a 4th case that pushes into an `observationOther` array, and display accordingly under a "Other Observations" section - just so we're not losing data in case there is another observation type

2. also, checking for the values "CCD Vitals", "CCD Social History", or "laboratory", explicitly in the appropriate property would be much more efficient than doing a stringify + string match - this is an expensive operation in comparison

let's at least address `1`, and put a TODO in for `2` to get this out the door